### DISCUSSION TITLE
How do I bind my data into it

### DISCUSSION THREAD

**Discussion by wosiyourfather** [AvaloniaUI/Avalonia#12641] [Q&A]
![9](https://github.com/AvaloniaUI/Avalonia/assets/53590491/8484affc-302a-4cd4-846f-d08f7f2b6ed3)
![99](https://github.com/AvaloniaUI/Avalonia/assets/53590491/47ea1694-f7b3-4de4-829d-f5975003cec6)

**Comment by wosiyourfather** [comment]
I need to change the background color based on the content

**Reply by timunie** [reply]
use a converter

**Reply by wosiyourfather** [reply]
After using the converter, bind It's empty

**Comment by wosiyourfather** [comment]
> 使用转换器

After using the converter, bind It's empty

**Comment by wosiyourfather** [comment]
This is a dynamic column, and I need to determine the value of the field after each cell and convert it to color

**Comment by sccombp** [comment]
I replied in your other post as well but here is a simple example of binding to dynamic columns and using a converter to set the rows background

[AvaloniaDatagridBindingToDatatableColourConverter.zip](https://github.com/AvaloniaUI/Avalonia/files/12425026/AvaloniaDatagridBindingToDatatableColourConverter.zip)

Binding to the row, note how we set the converter (I'm being lazy and hardcoding it to the first column, don't do this in the real world)
![image](https://github.com/AvaloniaUI/Avalonia/assets/124018297/6af0e8a9-cfc8-46d2-875d-2f4d42949473)


The converter (using an int as the property being bound to)
![image](https://github.com/AvaloniaUI/Avalonia/assets/124018297/94c1eee5-3a42-447a-97d1-182a966617b4)

The result
![image](https://github.com/AvaloniaUI/Avalonia/assets/124018297/f2e8176e-e0e8-4804-b65e-4ca005588053)

**Reply by sccombp** [reply]
oops, there's a bug in my example code.
In the converter remove the line
``` csharp
return Brushes.White;
```

My bad, was in a rush :/

**Reply by wosiyourfather** [reply]
![4](https://github.com/AvaloniaUI/Avalonia/assets/53590491/9fc0823c-a32d-4a46-add3-d51ede3ce0c9)
![44](https://github.com/AvaloniaUI/Avalonia/assets/53590491/c9e1ff53-908b-42fb-98c4-784174a007f3)
![444](https://github.com/AvaloniaUI/Avalonia/
![4444](https://github.com/AvaloniaUI/Avalonia/assets/53590491/b7df8e5d-ce3a-4c3c-a454-8bfeda23100b)
assets/53590491/67982f16-e742-43e6-9bdd-8c3289410fd6)
[AvaloniaDatagridBindingToDatatable.zip](https://github.com/AvaloniaUI/Avalonia/files/12425163/AvaloniaDatagridBindingToDatatable.zip)
Thank you for your example, but I need each cell to present a different color.For example, in Winform, I can implement it in this way. In Avalonia, I set up such an entity. But it's not possible to bind every cell to this entity

**Reply by wosiyourfather** [reply]
![444](https://github.com/AvaloniaUI/Avalonia/assets/53590491/fdd742a9-641e-4079-afa5-c92a85012066)
Because of the dynamic column case, I also need to set the background color of each cell

**Reply by wosiyourfather** [reply]
Please help me look at this example
[AvaloniaDatagridBindingToDatatable.zip](https://github.com/AvaloniaUI/Avalonia/files/12425198/AvaloniaDatagridBindingToDatatable.zip)

**Comment by wosiyourfather** [comment]
> I replied in your other post as well but here is a simple example of binding to dynamic columns and using a converter to set the rows background
> 
> [AvaloniaDatagridBindingToDatatableColourConverter.zip](https://github.com/AvaloniaUI/Avalonia/files/12425026/AvaloniaDatagridBindingToDatatableColourConverter.zip)
> 
> Binding to the row, note how we set the converter (I'm being lazy and hardcoding it to the first column, don't do this in the real world) ![image](https://user-images.githubusercontent.com/124018297/262847029-6af0e8a9-cfc8-46d2-875d-2f4d42949473.png)
> 
> The converter (using an int as the property being bound to) ![image](https://user-images.githubusercontent.com/124018297/262847192-94c1eee5-3a42-447a-97d1-182a966617b4.png)
> 
> The result ![image](https://user-images.githubusercontent.com/124018297/262847416-f2e8176e-e0e8-4804-b65e-4ca005588053.png)
Thank you for your example, but I need each cell to present a different color.For example, in Winform, I can implement it in this way. In Avalonia, I set up such an entity. But it's not possible to bind every cell to this entity
[AvaloniaDatagridBindingToDatatable.zip](https://github.com/AvaloniaUI/Avalonia/files/12425158/AvaloniaDatagridBindingToDatatable.zip)

**Reply by timunie** [reply]
@wosiyourfather stop spamming the same question in several places please. This will not result in better support. Thank you 🙏

**Reply by wosiyourfather** [reply]
okey

**Reply by timunie** [reply]
Your sample is too complex for me. You are kind of re-inventing the wheel. If it is just about backgruound color of your cells, you can do it via CellStyle

```xml
<UserControl.Styles>
	<Style Selector="DataGridCell.MyColoredCell" >
		<Setter Property="Background" Value="{Binding Converter={StaticResource myConverter}}" />
	</Style>
</UserControl.Styles>
```

and in your column setup:

```diff
ExampleDatagrid.Columns.Add(new DataGridTextNewColumn()
{

    Header = x.HeadName,
    // CellTemplate = (DataTemplate)Resources["WikipediaImageCell"],
    Binding = new Avalonia.Data.Binding($"Cloumns"),
+   CellStyleClasses = { "MyColoredCell" }
    //Tag = new Avalonia.Data.Binding($"Row.ItemArray[{x.Ordinal}]"),
    //ClipboardContentBinding = new Avalonia.Data.Binding($"Row.ItemArray[{x.Ordinal}]")
});
```

**Reply by wosiyourfather** [reply]
Thank you, but how do I bind the values in the entity in this case? There are three fields in my dynamic column entity, one for setting color, one for setting background color, and the other for value

**Reply by wosiyourfather** [reply]
![ls](https://github.com/AvaloniaUI/Avalonia/assets/53590491/900780dd-5f6d-4f99-a21c-6f1d2671ef1a)
This is the effect I created using itemControl. I need this effect

**Reply by timunie** [reply]
same approach works for TemplateColumn or custom column...

I can't do more here, sorry.

**Reply by wosiyourfather** [reply]
Yes, just like there is a Rows. cells [] in Winform that can implement this method, but I couldn't find it in Avalonia

**Reply by timunie** [reply]
No Avalonia is completely different. You will need to learn new concepts comming from WinForms. You need to understand DataTemplates, Bindings, Converters etc before porting your product.

**Reply by wosiyourfather** [reply]
I used to do wpf, but in fact, listview can also achieve the desired effect very well, but unfortunately, there is no
